ui/raidboss/data/05-shb/alliance/the_tower_at_paradigms_breach.ts

```diff
@@ -468,6 +477,144 @@ const triggerSet: TriggerSet<Data> = {
+        } else if (matches.id === '5C03' || matches.id === '5C04') {
+          if (matches.heading === '1.570772')
+            data.deployArmaments.push(DeployArmaments.VerticalSides);
+          else if (matches.heading === '-4.792213E-05')
+            data.deployArmaments.push(DeployArmaments.HorizontalSides);
+        }
+      },
+    },
+    {
+      id: 'Paradigm Meng-Zi/Xun-Zi Deploy Armaments Trigger',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['5C00', '5C01', '5C03', '5C04'], capture: false }),
+      delaySeconds: 0.25,
+      durationSeconds: 5,
+      suppressSeconds: 0.5,
+      alertText: (data, _matches, output) => {
+        if (!data.deployArmaments)
+          return;
+
+        // define some expected results
+        const verticalMiddle = (1 << 1 | 1 << 4 | 1 << 7);
+        const verticalSides = (1 << 0 | 1 << 2 |
+          1 << 3 | 1 << 5 |
+          1 << 6 | 1 << 8);
+        const horizontalMiddle = (1 << 3 | 1 << 4 | 1 << 5);
+        const horizontalSides = (1 << 0 | 1 << 1 | 1 << 2 |
+          1 << 6 | 1 << 7 | 1 << 8);
+        const nsMiddle = (1 << 1 | 1 << 7);
+        const ewMiddle = (1 << 3 | 1 << 5);
+        const corners = (1 << 0 | 1 << 2 | 1 << 6 | 1 << 8);
+        const center = (1 << 4);
+
+        // Start with all spots marked as safe...
+        let safeSpots = (1 << 0 | 1 << 1 | 1 << 2 |
+          1 << 3 | 1 << 4 | 1 << 5 |
+          1 << 6 | 1 << 7 | 1 << 8);
+
+        // Then figure out which spots are unsafe due to pending AOEs...
+
+        if (data.deployArmaments.includes(DeployArmaments.VerticalMiddle)) {
+          // Clear the vertical middle column
+          safeSpots &= ~verticalMiddle;
+        }
+
+        if (data.deployArmaments.includes(DeployArmaments.VerticalSides)) {
+          // Clear the vertical side columns
+          safeSpots &= ~verticalSides;
+        }
+
+        if (data.deployArmaments.includes(DeployArmaments.HorizontalMiddle)) {
+          // Clear the horizontal middle row
+          safeSpots &= ~horizontalMiddle;
+        }
+
+        if (data.deployArmaments.includes(DeployArmaments.HorizontalSides)) {
+          // Clear the horizontal side rows
+          safeSpots &= ~horizontalSides;
+        }
+
+        // Finally, determine where to alert
+        if (
+          safeSpots === verticalMiddle ||
+          safeSpots === horizontalMiddle ||
+          safeSpots === center
+        )
+          return output.center!();
+        else if (safeSpots === verticalSides)
+          return output.ewSides!();
+        else if (safeSpots === horizontalSides)
+          return output.nsSides!();
+        else if (safeSpots === nsMiddle)
+          return output.nsMiddle!();
+        else if (safeSpots === ewMiddle)
+          return output.ewMiddle!();
+        else if (safeSpots === corners)
+          return output.corners!();
+
+        return output.oops!();
```

Comment:
The bitmask is a nice generalization, but it feels a bit like overkill and it's a little hard to follow and make sure everything is correct.  What do you think about only using a series of if statements?  I think if there were more cases, a bitmask approach would cut down on combinatoric conditionals, but in this case there's 1-2 abilities, and many combinations are mutually exclusive, so I don't think that there's that many cases to cover manually.

```suggestion
        if (!data.deployArmaments)
          return output.oops!();

        if (data.deployArmaments.includes(DeployArmaments.VerticalMiddle)) {
          if (data.deployArmaments.includes(DeployArmaments.HorizontalMiddle))
            return output.corners!();
          // etc...
        }
        // etc...
```

Re: callout.  I also am not sure I would be able to know what "E/W Middle" vs "E/W Sides" meant on the fly.  I think "E/W Middle" is a subset of "E/W Sides" if I squint?  If these two mobs always appear in the same place, what about using cardinal directions, e.g. "go west", or "go west middle" (?), or "go northwest", or "go center".  (Assuming they're north and west.  It probably makes sense to go between them for melee.)

Reply:
I think I agree. that we can stick to a simpler approach where we only care about the 4 corners closest to the two bosses. That seems reasonable. It's hard to call out "move east and stay in the middle"

The two enemies always are in either the north end of the platform, or the west end, never anywhere else. The only two that are a bit weird are the north middle and west middle areas. I'll see if I can come up with something.

Reply:
In regards to simpler setup, ya I think we can mostly make it easier by just simplifying the overall attempt to care about only 4 corners.

Reply:
I ended up with "Go to North Boss" and "Go to West Boss" for those spots. That makes the most sense. When only a single line is detected, it is "Go North" or "Go West".

Will have the new version up in a bit.